request-object/app.py

```diff
@@ -0,0 +1,34 @@
+# from flask import Flask, render_template, request
+from flask import *
+
+app = Flask(__name__)
+
+@app.route("/")
+def hello():
+    return render_template("hello.html")
+
+
+@app.route("/login", methods=["POST", "GET"])
+def login():
+    # None は値が存在しないことを示す定数
+    error = None
+    if request.method == "POST":
+        if valid_login(request.form["username"], request.form["password"]):
+            return log_the_user_in(request.form["username"])
+        else:
+            error = "Invalid username/password"
+    return render_template("login.html", error=error)
+
+
+def valid_login(username=None, password=None):
```

Comment:
[IMHO] why length check only?
I think, it is good to compare with hard coded values.
and add note.

``` python
def valid_login(username=None, password=None):
    # NOTE: 今はまだデータベースなどのシステムを使えなくユーザーの認証をできないので、適当にハードコーディングされた値と比較するようにする。
    if username == "noah" and password == "pass": # ご自身の好きなパラメータに書き換えてください
        return True
    return False
```